src/ngraph/builder/quantization_util.hpp

```diff
@@ -222,6 +222,60 @@ namespace ngraph
 
                 return max_abs_range / target_range;
             }
+
+            void check_concat(const NodeVector args,
+                              std::shared_ptr<Node> min,
+                              std::shared_ptr<Node> max)
+            {
+                auto type = min->get_element_type();
+                if (type != max->get_element_type())
+                {
+                    throw ngraph_error("check_concat: min and max must have same type");
+                }
+
+                auto shape = min->get_shape();
+                if (shape != max->get_shape())
+                {
+                    throw ngraph_error("check_concat: min and max must have same shape");
+                }
+
+                auto min_const_op = std::dynamic_pointer_cast<ngraph::op::Constant>(min);
+                auto max_const_op = std::dynamic_pointer_cast<ngraph::op::Constant>(max);
+                if (min_const_op == nullptr)
+                {
+                    throw ngraph_error("QuantizedConcat min must be a Constant");
+                }
+                if (max_const_op == nullptr)
+                {
+                    throw ngraph_error("QuantizedConcat max must be a Constant");
+                }
+
+                // TODO (nbpatel) The check below is mkldnn specific.
+                // Will need to sit in cpu_assignment pass probably.
```

Comment:
We could easily support the general case by adding a dequantize and quantize to each of the args that dont match the range of the first arg. 
arg1->Dequantize->Quantize(range of arg0)->arg1 for QuantizedConcat